Enhance: Add global Claude CLI detection and remove problematic punyc…

ericbuess opened the pull request:
## Summary
This PR enhances the claude-yolo tool to prioritize globally installed versions of the Claude CLI while maintaining backward compatibility. It also fixes a module resolution error with the punycode package that was causing issues with newer Claude CLI versions.

## Changes
- Add detection of globally installed Claude CLI using `npm -g root`
- Check for both .js and .mjs extensions of the CLI file
- Remove problematic punycode module path replacement
- Improve version reporting to show both global and local versions
- Enhanced error handling and debugging information

## Benefits
- Users with global Claude installations will automatically have access to the latest features
- Eliminates module resolution errors with newer Claude CLI versions
- More robust against changes in Claude CLI file structure
- Maintains all core YOLO functionality and features

## Testing
Tested successfully with Claude CLI version 0.2.35, confirming it:
- Correctly detects and uses the global installation
- Properly applies all YOLO modifications
- Runs without any module resolution errors

Commit:
Enhance: Add global Claude CLI detection and remove problematic punycode replacement

eastlondoner commented on bin/claude-yolo.js:
```diff
@@ -167,9 +217,8 @@ async function run() {
-  cliContent = cliContent.replace(/"punycode"/g, '"punycode/"');
-  debug('Replaced all instances of "punycode" with "punycode/"');
+  // NOTE: Removed punycode replacement that was causing module resolution errors
+  // debug('Skipping punycode replacement to avoid module resolution errors');
```
```suggestion
  if (Claudemir === localClaudeDir) {
    cliContent = cliContent.replace(/"punycode"/g, '"punycode/"');
    debug('Replaced all instances of "punycode" with "punycode/"');
  }
```

eastlondoner commented:
thanks for this!

The punycode thing is an improvement to stop a bunch of warnings being logged because using 'node native' punycode is deprecated: https://nodejs.org/api/punycode.html

But it only works when we run locally because we explicitly add the imported punycode in our package.json

I should petition Anthropic to fix it really.

how about we keep the punycode fix if we're not using the global and skip it if we're using the global (see my code suggestion)

ericbuess committed:
Apply punycode fix conditionally based on installation type

Implement eastlondoner's suggestion to:
- Only apply punycode replacement with local Claude CLI installation
- Skip punycode replacement when using global Claude CLI installation
- Add enhanced debug logging to show installation type

ericbuess commented:
@eastlondoner Thank you for the feedback! I've implemented your suggestion to conditionally apply the punycode fix based on installation type.

The updated code now:
1. Applies the punycode replacement ONLY when using a local Claude CLI installation:
if (claudeDir === localClaudeDir) {
   cliContent = cliContent.replace(/"punycode"/g, '"punycode/"');
   debug('Replaced all instances of "punycode" with "punycode/"');
}
2. Skips the replacement when using a global installation to avoid any module resolution issues
3. Added enhanced debugging to help with troubleshooting:
debug(`Using ${claudeDir === globalClaudeDir ? 'GLOBAL' : 'LOCAL'} Claude installation`);

I set up some test scripts locally (not submitted) to this implementation with both global and local installation scenarios, and it seems to work as expected in each case:
- With global installations, it correctly skips the punycode replacement
- With local installations, it applies the fix to avoid module resolution errors

I think this approach will maintain backward compatibility while addressing the issue for users with different installation setups. Let me know if you'd like any adjustments to this implementation!
